### DISCUSSION TITLE
Example using <router-outlet> inside <ngt-canvas>

### DISCUSSION THREAD

**Discussion by IRobot1** [angular-threejs/angular-three#2] [General]
Can you provide an example of using router-outlet inside ngt-canvas where the scene is defined by the routed child component.

```html
  <ngt-canvas>
    <router-outlet></router-outlet>
  </ngt-canvas>
```

**Comment by nartc** [comment]
Hm... I think I overlooked this use-case. I need to check if/how this works. Thanks for bringing it up

**Comment by IRobot1** [comment]
Will it be possible to create the canvas once and just change the scene via the route?  

I saw [this post ](https://discourse.threejs.org/t/how-should-i-handle-webgl-context-when-switching-to-a-new-page/42018/2) recently that creating and destroying canvases on page changes isn't great and will lead to problems.

**Comment by nartc** [comment]
I came up with something and I'm not sure if it's good enough. Here's the gist:

<img width="1141" alt="image" src="https://user-images.githubusercontent.com/25516557/219543057-026b6f57-060a-40b0-ab07-c365ed2f4287.png">
Provide your routes

<img width="917" alt="image" src="https://user-images.githubusercontent.com/25516557/219543111-4ca9d347-bd86-4e5c-a63e-0c9bf9c5d246.png">
Then use `NgtRoutedScene` which is a component from `angular-three`

<img width="1730" alt="image" src="https://user-images.githubusercontent.com/25516557/219543145-1ef340d5-7534-42e0-b510-22f1e36e59c2.png">
<img width="1709" alt="image" src="https://user-images.githubusercontent.com/25516557/219543157-31908a13-ed7e-4221-bcfe-8b93b0ef8439.png">
Routed, and lazy-load
